providers/base/bin/wifi_nmcli_test.py

```diff
@@ -351,29 +411,45 @@ def print_journal_entries(start):
 
     args = parser.parse_args()
 
-    start_time = datetime.datetime.now()
+    return args
+
 
-    cleanup_nm_connections()
+def main():
+    args = parser_args()
+    start_time = datetime.datetime.now()
     device_rescan()
-    count = list_aps(args)
+    essid = getattr(args, "essid", None)
+    aps_dict = list_aps(args.device, essid)
+    show_aps(aps_dict)
 
     if args.test_type == "scan":
-        if count == 0:
+        if not aps_dict:
             print("Failed to find any APs")
-            sys.exit(1)
+            return 1
         else:
-            print("Found {} access points".format(count))
-            sys.exit(0)
+            print("Found {} access points".format(len(aps_dict)))
+            return 0
+
+    if not aps_dict:
+        print("Targed access points: {} not found".format(args.essid))
+        return 1
 
     if args.func:
+        delete_test_ap_ssid_connection()
+        activated_uuid = get_nm_activate_connection()
+        turn_down_nm_connections()
         try:
             result = args.func(args)
```

Comment:
I know this is not part of this PR, but why is the function a part of args?
Wouldn't it be clear and easier to test if the function to use was defined in `main()` instead of in the parser, as you are doing with the scan?
```suggestion
        try:
            if args.test_type == parser_open:
                result = open_connection(args.device, args.essid)
```

Reply:
In the script support two different type of wifi test.
One is the wifi station mode (connection to ap); another is wifi AP mode (to be a ap)

```
parser_secured.set_defaults(func=secured_connection)
parser_ap.set_defaults(func=hotspot)
```

Those you mention will be in the next PR.